Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -397,6 +402,10 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
 
         if (urlWasBlocked != urlIsBlocked)
             [self setAllMediaPlaybackSuspended:urlIsBlocked completionHandler:nil];
+
+        [self willChangeValueForKey:@"_isBlockedByScreenTime"];
+        _isBlockedByScreenTime = urlIsBlocked;
+        [self didChangeValueForKey:@"_isBlockedByScreenTime"];
```

Comment:
This should probably be inside the `if (urlWasBlocked != urlIsBlocked)`?

Reply:
(this comment still hasn't been addressed)

Reply:
So for this, I noticed the`IsBlockedByScreenTimeKVOFalse` would fail if this code block was inside the `if`. This is b/c we start out with url _not_ blocked, then set it to url _not_ blocked again, so `_isBlockedByScreenTime` never changes and says it never updates. I was going to ask if I should 1) keep the current or 2) make the change and remove that test (I could also set urlIsBlocked to T then F but it would be basically identical to `IsBlockedByScreenTimeKVOTrue`.

Reply:
> This is b/c we start out with url not blocked, then set it to url not blocked again, so _isBlockedByScreenTime never changes and says it never updates

In this case, it is expected that no updates are fired. So `IsBlockedByScreenTimeKVOFalse` is an incorrect test.

> (I could also set urlIsBlocked to T then F but it would be basically identical to IsBlockedByScreenTimeKVOTrue)

This test is still valuable as you'd be testing the T -> F transition, which is currently untested.

Reply:
Ok sounds good! Will change!

Reply:
(you could just update `IsBlockedByScreenTimeKVOTrue` to test the T -> F transition right after the F -> T transition, and rename the test to `IsBlockedByScreenTimeKVO`)